services/workers/src/test/java/org/sagebionetworks/report/worker/StorageReportCSVDownloadWorkerTest.java

```diff
@@ -0,0 +1,131 @@
+import org.sagebionetworks.repo.manager.file.LocalFileUploadRequest;
+import org.sagebionetworks.repo.manager.report.StorageReportManager;
+import org.sagebionetworks.repo.model.AuthorizationConstants.BOOTSTRAP_PRINCIPAL;
+import org.sagebionetworks.repo.model.UserInfo;
+import org.sagebionetworks.repo.model.asynch.AsynchronousJobStatus;
+import org.sagebionetworks.repo.model.dbo.dao.table.TableExceptionTranslator;
+import org.sagebionetworks.repo.model.file.S3FileHandle;
+import org.sagebionetworks.repo.model.report.DownloadStorageReportRequest;
+import org.sagebionetworks.repo.model.report.DownloadStorageReportResponse;
+import org.sagebionetworks.repo.model.report.StorageReportType;
+import org.sagebionetworks.util.Clock;
+import org.sagebionetworks.util.csv.CSVWriterStream;
+import org.sagebionetworks.workers.util.aws.message.RecoverableMessageException;
+import org.sagebionetworks.workers.util.semaphore.LockUnavilableException;
+import org.springframework.test.util.ReflectionTestUtils;
+
+import com.amazonaws.services.sqs.model.Message;
+
+
+@RunWith(MockitoJUnitRunner.class)
+public class StorageReportCSVDownloadWorkerTest {
+	
+	public static final long MAX_WAIT_MS = 1000 * 60;
+
+	private StorageReportCSVDownloadWorker storageReportWorker;
+	@Mock
+	private UserManager mockUserManager;
+	@Mock
+	private StorageReportManager mockStorageReportManager;
+	@Mock
+	private AsynchJobStatusManager mockAsyncMgr;
+	@Mock
+	private AsynchronousJobStatus mockStatus;
+	@Mock
+	private FileHandleManager mockFileHandleManager;
+	@Mock
+	private Clock mockClock;
+	@Mock
+	private TableExceptionTranslator mockTableExceptionTranslator;
+
+	DownloadStorageReportRequest request;
+	UserInfo adminUser;
+	Long adminUserId = BOOTSTRAP_PRINCIPAL.THE_ADMIN_USER.getPrincipalId();
+	S3FileHandle resultS3FileHandle;
+	String resultS3FileHandleId = "999";
+	private static final Message message = new Message();
+
+	@Before
+	public void before(){
+
+		request = new DownloadStorageReportRequest();
+		request.setReportType(StorageReportType.ALL_PROJECTS);
+		adminUser = new UserInfo(true);
+		adminUser.setId(adminUserId);
+		storageReportWorker = new StorageReportCSVDownloadWorker();
+
+		resultS3FileHandle = new S3FileHandle();
+		resultS3FileHandle.setId(resultS3FileHandleId);
+		when(mockFileHandleManager.multipartUploadLocalFile(any(LocalFileUploadRequest.class))).thenReturn(resultS3FileHandle);
+		ReflectionTestUtils.setField(storageReportWorker, "storageReportManager", mockStorageReportManager);
+		ReflectionTestUtils.setField(storageReportWorker, "asynchJobStatusManager", mockAsyncMgr);
+		ReflectionTestUtils.setField(storageReportWorker, "userManager", mockUserManager);
+		ReflectionTestUtils.setField(storageReportWorker, "fileHandleManager", mockFileHandleManager);
+		ReflectionTestUtils.setField(storageReportWorker, "clock", mockClock);
+		ReflectionTestUtils.setField(storageReportWorker, "tableExceptionTranslator", mockTableExceptionTranslator);
+	}
+
```

Comment:
Add file provider to the class under test and then test it here to ensure the temp file is deleted